File: common/src/main/java/com/pluxity/global/response/BaseResponse.java
package com.pluxity.global.response;

import com.pluxity.global.entity.BaseEntity;

public record BaseResponse(String createdAt, String createdBy, String updatedAt, String updatedBy) {

    public static BaseResponse of(BaseEntity entity) {
        return new BaseResponse(
                entity.getCreatedAt() != null ? entity.getCreatedAt().toString() : null,
                entity.getCreatedBy() != null ? entity.getCreatedBy() : null,
                entity.getUpdatedAt() != null ? entity.getUpdatedAt().toString() : null,
                entity.getUpdatedBy() != null ? entity.getUpdatedBy() : null);
    }
}

File: common/src/main/kotlin/com/pluxity/global/response/BaseResponse.kt
package com.pluxity.global.response

import com.pluxity.global.entity.BaseEntity

data class BaseResponse(
    val createdAt: String,
    val createdBy: String,
    val updatedAt: String,
    val updatedBy: String,
)

fun BaseEntity.toBaseResponse(): BaseResponse =
    BaseResponse(
        createdAt = this.createdAt.toString(),
        createdBy = this.createdBy!!,
        updatedAt = this.updatedAt.toString(),
        updatedBy = this.updatedBy!!,
    )
